code401challenges/src/main/java/linkedList/LinkedList.java

```diff
@@ -55,4 +55,71 @@ public String toString() {
 
         return sb.toString();
     }
+
+    // append method
+    // adds a new node with the given value to the end of the list
+    public void append(T value) {
+        Node<T> current = head;
+        Node<T> newNode = new Node<>(value);
+
+        if (current == null) {
+            head = newNode;
+            return;
+        }
+
+        while(current != null) {
+            if (current.next == null) {
+                // append new node
+                current.next = newNode;
+                return;
+            }
+
+            current = current.next;
+        }
+    }
+
+    public void insertBefore(T value, T newVal) {
+        Node<T> current = head;
+        Node<T> newNode = new Node<>(newVal);
+
+        if (current == null) {
```

Comment:
Food for thought: If `current` is `null` then that means `value` isn't in the linked list. If `value` isn't in the list should `newVal` be inserted at all?